.github/workflows/codespell.yml

```diff
@@ -0,0 +1,25 @@
+# Codespell configuration is within .codespellrc
+---
+name: Codespell
+
+on:
+  push:
+    branches: [master]
+  pull_request:
+    branches: [master]
```

Comment:
```suggestion
  [push, pull_request]
```

Reply:
we could do that, but likely would need to get to the same line as `on: `?

Reply:
but also it would lead to double runs I think -- it might then run for the PR branch (e.g. "blah") and also then for the PR itself... adding noise etc.  So -- are you sure? what are you trying to achieve with widening the scope of when to run?

Reply:
I would want tests to run when we make PRs from feature branches to our dev branch (we don't merge features directly to master) or between our feature branches.

Reply:
so let's add `dev` there?
```suggestion
  push:
    branches: [master, dev]
  pull_request:
    branches: [master, dev]
```
so that pushes and PRs against `dev` are also tested?

Reply:
That wouldn't include feature->feature PRs.